### PR TITLE
add tls cleanup toprotect udp

### PR DESCRIPTION
**Pull request by wangchen61698**
## Summary
Due to abnormal application exit,so add tls cleanup to protect udp process
## Impact
udp
## Testing
Manual verification

### CODE REVIEW

File: net/udp/udp.h [hunk 1]
```diff
@@ -175,6 +175,14 @@ struct udp_wrbuffer_s
 };
 #endif
 
+struct udp_callback_s
+{
+  FAR struct net_driver_s *dev;
+  FAR struct udp_conn_s *conn;
+  FAR struct devif_callback_s *udp_cb;
+  sem_t *sem;
```
Comment (pkarashchenko) [inline net/udp/udp.h:183]: Please add `FAR`

Comment (wangchen61698) [reply net/udp/udp.h]: done

**Comment by anchao** [comment]
@xiaoxiang781216 
The life cycle of the socket should be managed with file descriptor. If an abnormal exit causes an exception, then the VFS needs to be further enhanced. If all subsystems depends on TLS when exiting, then the entire kernel will be surrounded by TLS. The implementation of the protocol stack should further learn from the Linux kernel instead of applying these ugly patches!

**Comment by anchao** [comment]
If similar changes are allowed, in the future any waiting logic in the subsystem will need to depends on `tls_cleanup_push`/`tls_cleanup_pop`, Is this what you expect?

**Comment by wengzhe** [comment]
> @xiaoxiang781216 The life cycle of the socket should be managed with file descriptor. If an abnormal exit causes an exception, then the VFS needs to be further enhanced. If all subsystems depends on TLS when exiting, then the entire kernel will be surrounded by TLS. The implementation of the protocol stack should further learn from the Linux kernel instead of applying these ugly patches!

@anchao Agree, and maybe we'll not merge this PR into the master. BTW, is `CONFIG_CANCELLATION_POINTS` well tested? Socket operations are already protected by cancellation points, but we haven't enabled it on products yet.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by anchao — APPROVED** [review]

**Pull request merged**